Make the output of the dot format deterministic and stable

kdeldycke opened the pull request:
This is an adhoc re-interpretation of the [`graphviz.dot.Dot.__iter__()` method](https://github.com/xflr6/graphviz/blob/master/graphviz/dot.py#L158-L182) that is used behind the scene by [the original `graph.source` call](https://github.com/tox-dev/pipdeptree/blob/2eb489449169b59711a4fc979f4fc3483e8e81b0/src/pipdeptree/__init__.py#L613-L615) to dump the graphviz dot file content in `pipdeptree` (i.e. when the `--graph-output dot` parameter in called on the CLI).

But in this PR, we forces the sorting of the generated nodes and edge statements of the graph. That way we address issue #188, in which each invokation of the `pipdeptree` generates a different dot file.

The order of the `dot`-statements does not matter because of the nature of dependencies and the way `pipdeptree` process them. That's why a call to `sorted()` works without any special care. Graphviz will anyway re-interpret internally the relationships on rendering.

gaborbernat reviewed (changes requested):
Fix CI, rebase on main and add test.

kdeldycke commented:
Just rebased, squashed, fixed and added unit-tests to ensure stability of Graphviz output.

All tests are passing, this PR is ready to be merged upstream.

kdeldycke commented:
Thanks @gaborbernat for the merge!
